### ISSUE TITLE
pylibcugraphops install fail

### ISSUE THREAD

**Issue by paoxiaode** [dmlc/dgl#5672]
## 🐛 Bug

Try to use the cugraph op in v1.1, I try to install pylibcugraphops but fail.

https://docs.dgl.ai/generated/dgl.nn.pytorch.conv.CuGraphGATConv.html

## To Reproduce

conda create --name cugraph python=3.10
conda activate cugraph
conda install -c nvidia pylibcugraphops

![image](https://github.com/dmlc/dgl/assets/51984235/075b99f4-36af-4535-bdaf-c0b121a22f1d)

## Environment

 - DGL Version (e.g., 1.1):1,1
 - Backend Library & Version (e.g., PyTorch 0.4.1, MXNet/Gluon 1.3): 2.0.1python
 - OS (e.g., Linux): Linux
 - How you installed DGL (`conda`, `pip`, source): pip
 - Python version: 3.10
 - CUDA/cuDNN version (if applicable): 11.7

**Comment by Rhett-Ying** [comment]
@tingyu66 could you help take a look at this issue?

**Comment by Rhett-Ying** [comment]
@tingyu66 `conda install -c nvidia -c conda-forge -c rapidsai pylibcugraphops` does not work for me. I am working on Ubuntu20.04, dgl 1.1.0+cu117, python 3.10, torch 2.0.0+cu117
```
Package ca-certificates conflicts for:
openssl -> ca-certificates
python=3.10 -> openssl[version='>=3.1.0,<4.0a0'] -> ca-certificates
ca-certificatesThe following specifications were found to be incompatible with your system:

  - feature:/linux-64::__glibc==2.31=0
  - feature:|@/linux-64::__glibc==2.31=0
  - pylibcugraphops -> __glibc[version='>=2.17,<3.0.a0']
  - pylibcugraphops -> librmm[version='>=22.12.0,<22.13.0a0'] -> __glibc[version='>=2.17']

Your installed version is: 2.31

Note that strict channel priority may have removed packages required for satisfiability.

(dgl-dev-gpu) ubuntu@ip-172-31-51-7:~/workspace/dgl_0$ ipython3
Python 3.10.10 | packaged by conda-forge | (main, Mar 24 2023, 20:08:06) [GCC 11.3.0]
Type 'copyright', 'credits' or 'license' for more information
IPython 8.13.2 -- An enhanced Interactive Python. Type '?' for help.

In [1]: import dgl
   ...: import torch
   ...: from dgl.nn import CuGraphGATConv
   ...: device = 'cuda'
   ...: g = dgl.graph(([0,1,2,3,2,5], [1,2,3,4,0,3])).to(device)
   ...: g = dgl.add_self_loop(g)
   ...: feat = torch.ones(6, 10).to(device)
   ...: conv = CuGraphGATConv(10, 2, num_heads=3).to(device)
   ...: res = conv(g, feat)
   ...: res
---------------------------------------------------------------------------
ModuleNotFoundError                       Traceback (most recent call last)
Cell In[1], line 8
      6 g = dgl.add_self_loop(g)
      7 feat = torch.ones(6, 10).to(device)
----> 8 conv = CuGraphGATConv(10, 2, num_heads=3).to(device)
      9 res = conv(g, feat)
     10 res

File /opt/conda/envs/dgl-dev-gpu/lib/python3.10/site-packages/dgl/nn/pytorch/conv/cugraph_gatconv.py:95, in CuGraphGATConv.__init__(self, in_feats, out_feats, num_heads, feat_drop, negative_slope, residual, activation, bias)
     83 def __init__(
     84     self,
     85     in_feats,
   (...)
     92     bias=True,
     93 ):
     94     if has_pylibcugraphops is False:
---> 95         raise ModuleNotFoundError(
     96             f"{self.__class__.__name__} requires pylibcugraphops >= 23.02. "
     97             f"Install via `conda install -c nvidia 'pylibcugraphops>=23.02'`."
     98         )
     99     super().__init__()
    100     self.in_feats = in_feats

ModuleNotFoundError: CuGraphGATConv requires pylibcugraphops >= 23.02. Install via `conda install -c nvidia 'pylibcugraphops>=23.02'`.

```

**Comment by tingyu66** [comment]
@Rhett-Ying I just tried it on my end and it works. I attach the full history for your reference.

```bash
➜  dgl git:(master) ✗ bash script/create_dev_conda_env.sh -g 11.7 -p 3.10 -t 2.0.0
Confirm the installed CUDA version matches the specified one.
Continue? [yes/no]:
yes
Confirm you are excuting the script from your DGL root directory.
Current working directory: /home/tingyuw/git/dgl
Continue? [yes/no]:
yes
--------------------------------------------------
name: dgl-dev-gpu
dependencies:
  - python=3.10
  - pip
  - graphviz
  - pandoc
  - pygraphviz
  - pip:
    - --find-links https://download.pytorch.org/whl/torch_stable.html
    - cython
    - filelock
    - matplotlib
    - networkx
    - nltk
    - nose
    - numpy
    - ogb
    - pandas
    - psutil
    - pyarrow
    - pydantic
    - pytest
    - pyyaml
    - rdflib
    - requests[security]
    - scikit-learn
    - scipy
    - torch==2.0.0+cu117
    - torchmetrics
    - tqdm
    - boto3 # AWS SDK for python
    - sphinx==4.2.0
    - sphinx-gallery
    - sphinx_rtd_theme
    - sphinx_copybutton
    - sphinxemoji
    - nbsphinx>=0.8.11
    - nbsphinx-link>=1.3.0
    - pillow
    - seaborn
    - jupyter_http_over_ws
variables:
  DGL_HOME: /home/tingyuw/git/dgl
--------------------------------------------------
Create a conda enviroment with the config?
Continue? [yes/no]:
yes
Collecting package metadata (repodata.json): done
Solving environment: done


==> WARNING: A newer version of conda exists. <==
  current version: 22.9.0
  latest version: 23.3.1

Please update conda by running

    $ conda update -n base -c defaults conda



Downloading and Extracting Packages
boost-cpp-1.73.0     | 16 KB     | ################################################################################################################### | 100% 
librsvg-2.54.4       | 6.0 MB    | ################################################################################################################### | 100% 
pandoc-2.12          | 10.5 MB   | ################################################################################################################### | 100% 
font-ttf-source-code | 654 KB    | ################################################################################################################### | 100% 
jpeg-9e              | 262 KB    | ################################################################################################################### | 100% 
pango-1.50.7         | 427 KB    | ################################################################################################################### | 100% 
font-ttf-ubuntu-0.83 | 1.5 MB    | ################################################################################################################### | 100% 
libboost-1.73.0      | 13.8 MB   | ################################################################################################################### | 100% 
libdeflate-1.17      | 69 KB     | ################################################################################################################### | 100% 
libxml2-2.10.3       | 755 KB    | ################################################################################################################### | 100% 
gobject-introspectio | 1.6 MB    | ################################################################################################################### | 100% 
fontconfig-2.14.1    | 281 KB    | ################################################################################################################### | 100% 
ninja-1.10.2         | 8 KB      | ################################################################################################################### | 100% 
libwebp-1.2.4        | 86 KB     | ################################################################################################################### | 100% 
openjpeg-2.4.0       | 331 KB    | ################################################################################################################### | 100% 
poppler-22.12.0      | 17.3 MB   | ################################################################################################################### | 100% 
cairo-1.16.0         | 1.4 MB    | ################################################################################################################### | 100% 
font-ttf-inconsolata | 83 KB     | ################################################################################################################### | 100% 
atk-1.0-2.36.0       | 466 KB    | ################################################################################################################### | 100% 
graphviz-2.50.0      | 2.6 MB    | ################################################################################################################### | 100% 
giflib-5.2.1         | 80 KB     | ################################################################################################################### | 100% 
c-ares-1.19.0        | 118 KB    | ################################################################################################################### | 100% 
ninja-base-1.10.2    | 109 KB    | ################################################################################################################### | 100% 
gdk-pixbuf-2.42.10   | 600 KB    | ################################################################################################################### | 100% 
poppler-data-0.4.11  | 2.6 MB    | ################################################################################################################### | 100% 
libcurl-7.88.1       | 383 KB    | ################################################################################################################### | 100% 
harfbuzz-4.3.0       | 1.3 MB    | ################################################################################################################### | 100% 
fribidi-1.0.10       | 103 KB    | ################################################################################################################### | 100% 
pygraphviz-1.9       | 136 KB    | ################################################################################################################### | 100% 
libgd-2.3.3          | 240 KB    | ################################################################################################################### | 100% 
Preparing transaction: done
Verifying transaction: done
Executing transaction: done
Installing pip dependencies: | Ran pip subprocess with arguments:
['/home/tingyuw/miniconda3/envs/dgl-dev-gpu/bin/python', '-m', 'pip', 'install', '-U', '-r', '/tmp/e2222e1ed0dc0c703fca/condaenv.49vomnc9.requirements.txt']
Pip subprocess output:
Looking in links: https://download.pytorch.org/whl/torch_stable.html
Collecting cython
  Using cached Cython-0.29.34-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.manylinux_2_24_x86_64.whl (1.9 MB)
Collecting filelock
  Downloading filelock-3.12.0-py3-none-any.whl (10 kB)
Collecting matplotlib
  Using cached matplotlib-3.7.1-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (11.6 MB)
Collecting networkx
  Downloading networkx-3.1-py3-none-any.whl (2.1 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 2.1/2.1 MB 19.9 MB/s eta 0:00:00
Collecting nltk
  Downloading nltk-3.8.1-py3-none-any.whl (1.5 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 1.5/1.5 MB 30.3 MB/s eta 0:00:00
Collecting nose
  Downloading nose-1.3.7-py3-none-any.whl (154 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 154.7/154.7 kB 18.5 MB/s eta 0:00:00
Collecting numpy
  Downloading numpy-1.24.3-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (17.3 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 17.3/17.3 MB 43.2 MB/s eta 0:00:00
Collecting ogb
  Downloading ogb-1.3.6-py3-none-any.whl (78 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 78.8/78.8 kB 9.6 MB/s eta 0:00:00
Collecting pandas
  Downloading pandas-2.0.1-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (12.3 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 12.3/12.3 MB 48.9 MB/s eta 0:00:00
Collecting psutil
  Downloading psutil-5.9.5-cp36-abi3-manylinux_2_12_x86_64.manylinux2010_x86_64.manylinux_2_17_x86_64.manylinux2014_x86_64.whl (282 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 282.1/282.1 kB 30.9 MB/s eta 0:00:00
Collecting pyarrow
  Downloading pyarrow-12.0.0-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (38.9 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 38.9/38.9 MB 36.2 MB/s eta 0:00:00
Collecting pydantic
  Downloading pydantic-1.10.7-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (3.1 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 3.1/3.1 MB 43.2 MB/s eta 0:00:00
Collecting pytest
  Downloading pytest-7.3.1-py3-none-any.whl (320 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 320.5/320.5 kB 35.0 MB/s eta 0:00:00
Collecting pyyaml
  Using cached PyYAML-6.0-cp310-cp310-manylinux_2_5_x86_64.manylinux1_x86_64.manylinux_2_12_x86_64.manylinux2010_x86_64.whl (682 kB)
Collecting rdflib
  Using cached rdflib-6.3.2-py3-none-any.whl (528 kB)
Collecting requests[security]
  Downloading requests-2.30.0-py3-none-any.whl (62 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 62.5/62.5 kB 7.1 MB/s eta 0:00:00
Collecting scikit-learn
  Using cached scikit_learn-1.2.2-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (9.6 MB)
Collecting scipy
  Using cached scipy-1.10.1-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (34.4 MB)
Collecting torch==2.0.0+cu117
  Downloading https://download.pytorch.org/whl/cu117/torch-2.0.0%2Bcu117-cp310-cp310-linux_x86_64.whl (1843.9 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 1.8/1.8 GB 3.8 MB/s eta 0:00:00
Collecting torchmetrics
  Using cached torchmetrics-0.11.4-py3-none-any.whl (519 kB)
Collecting tqdm
  Using cached tqdm-4.65.0-py3-none-any.whl (77 kB)
Collecting boto3
  Downloading boto3-1.26.132-py3-none-any.whl (135 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 135.6/135.6 kB 12.1 MB/s eta 0:00:00
Collecting sphinx==4.2.0
  Downloading Sphinx-4.2.0-py3-none-any.whl (3.1 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 3.1/3.1 MB 45.8 MB/s eta 0:00:00
Collecting sphinx-gallery
  Downloading sphinx_gallery-0.13.0-py3-none-any.whl (147 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 147.4/147.4 kB 17.6 MB/s eta 0:00:00
Collecting sphinx_rtd_theme
  Downloading sphinx_rtd_theme-1.2.0-py2.py3-none-any.whl (2.8 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 2.8/2.8 MB 43.1 MB/s eta 0:00:00
Collecting sphinx_copybutton
  Downloading sphinx_copybutton-0.5.2-py3-none-any.whl (13 kB)
Collecting sphinxemoji
  Using cached sphinxemoji-0.2.0.tar.gz (44 kB)
  Preparing metadata (setup.py): started
  Preparing metadata (setup.py): finished with status 'done'
Collecting nbsphinx>=0.8.11
  Downloading nbsphinx-0.9.1-py3-none-any.whl (30 kB)
Collecting nbsphinx-link>=1.3.0
  Using cached nbsphinx_link-1.3.0-py2.py3-none-any.whl (5.2 kB)
Collecting pillow
  Downloading Pillow-9.5.0-cp310-cp310-manylinux_2_28_x86_64.whl (3.4 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 3.4/3.4 MB 46.1 MB/s eta 0:00:00
Collecting seaborn
  Downloading seaborn-0.12.2-py3-none-any.whl (293 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 293.3/293.3 kB 33.7 MB/s eta 0:00:00
Collecting jupyter_http_over_ws
  Downloading jupyter_http_over_ws-0.0.8-py2.py3-none-any.whl (18 kB)
Collecting triton==2.0.0
  Downloading triton-2.0.0-1-cp310-cp310-manylinux2014_x86_64.manylinux_2_17_x86_64.whl (63.3 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 63.3/63.3 MB 36.1 MB/s eta 0:00:00
Collecting sympy
  Downloading sympy-1.12-py3-none-any.whl (5.7 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 5.7/5.7 MB 54.0 MB/s eta 0:00:00
Collecting typing-extensions
  Using cached typing_extensions-4.5.0-py3-none-any.whl (27 kB)
Collecting jinja2
  Downloading Jinja2-3.1.2-py3-none-any.whl (133 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 133.1/133.1 kB 16.6 MB/s eta 0:00:00
Collecting sphinxcontrib-devhelp
  Using cached sphinxcontrib_devhelp-1.0.2-py2.py3-none-any.whl (84 kB)
Collecting babel>=1.3
  Downloading Babel-2.12.1-py3-none-any.whl (10.1 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 10.1/10.1 MB 47.4 MB/s eta 0:00:00
Collecting sphinxcontrib-serializinghtml>=1.1.5
  Using cached sphinxcontrib_serializinghtml-1.1.5-py2.py3-none-any.whl (94 kB)
Collecting alabaster<0.8,>=0.7
  Using cached alabaster-0.7.13-py3-none-any.whl (13 kB)
Collecting sphinxcontrib-applehelp
  Using cached sphinxcontrib_applehelp-1.0.4-py3-none-any.whl (120 kB)
Requirement already satisfied: setuptools in /home/tingyuw/miniconda3/envs/dgl-dev-gpu/lib/python3.10/site-packages (from sphinx==4.2.0->-r /tmp/e2222e1ed0dc0c703fca/condaenv.49vomnc9.requirements.txt (line 24)) (66.0.0)
Collecting docutils<0.18,>=0.14
  Using cached docutils-0.17.1-py2.py3-none-any.whl (575 kB)
Collecting packaging
  Downloading packaging-23.1-py3-none-any.whl (48 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 48.9/48.9 kB 5.4 MB/s eta 0:00:00
Collecting sphinxcontrib-htmlhelp>=2.0.0
  Using cached sphinxcontrib_htmlhelp-2.0.1-py3-none-any.whl (99 kB)
Collecting sphinxcontrib-qthelp
  Using cached sphinxcontrib_qthelp-1.0.3-py2.py3-none-any.whl (90 kB)
Collecting snowballstemmer>=1.1
  Using cached snowballstemmer-2.2.0-py2.py3-none-any.whl (93 kB)
Collecting sphinxcontrib-jsmath
  Using cached sphinxcontrib_jsmath-1.0.1-py2.py3-none-any.whl (5.1 kB)
Collecting Pygments>=2.0
  Downloading Pygments-2.15.1-py3-none-any.whl (1.1 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 1.1/1.1 MB 43.8 MB/s eta 0:00:00
Collecting imagesize
  Using cached imagesize-1.4.1-py2.py3-none-any.whl (8.8 kB)
Collecting lit
  Downloading lit-16.0.3.tar.gz (138 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 138.0/138.0 kB 16.9 MB/s eta 0:00:00
  Preparing metadata (setup.py): started
  Preparing metadata (setup.py): finished with status 'done'
Collecting cmake
  Downloading cmake-3.26.3-py2.py3-none-manylinux2014_x86_64.manylinux_2_17_x86_64.whl (24.0 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 24.0/24.0 MB 43.1 MB/s eta 0:00:00
Collecting contourpy>=1.0.1
  Using cached contourpy-1.0.7-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (300 kB)
Collecting python-dateutil>=2.7
  Downloading python_dateutil-2.8.2-py2.py3-none-any.whl (247 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 247.7/247.7 kB 29.0 MB/s eta 0:00:00
Collecting pyparsing>=2.3.1
  Using cached pyparsing-3.0.9-py3-none-any.whl (98 kB)
Collecting kiwisolver>=1.0.1
  Using cached kiwisolver-1.4.4-cp310-cp310-manylinux_2_12_x86_64.manylinux2010_x86_64.whl (1.6 MB)
Collecting cycler>=0.10
  Using cached cycler-0.11.0-py3-none-any.whl (6.4 kB)
Collecting fonttools>=4.22.0
  Downloading fonttools-4.39.4-py3-none-any.whl (1.0 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 1.0/1.0 MB 53.7 MB/s eta 0:00:00
Collecting regex>=2021.8.3
  Downloading regex-2023.5.5-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (769 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 769.7/769.7 kB 44.7 MB/s eta 0:00:00
Collecting click
  Using cached click-8.1.3-py3-none-any.whl (96 kB)
Collecting joblib
  Using cached joblib-1.2.0-py3-none-any.whl (297 kB)
Collecting outdated>=0.2.0
  Using cached outdated-0.2.2-py2.py3-none-any.whl (7.5 kB)
Collecting urllib3>=1.24.0
  Downloading urllib3-2.0.2-py3-none-any.whl (123 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 123.2/123.2 kB 14.8 MB/s eta 0:00:00
Collecting six>=1.12.0
  Downloading six-1.16.0-py2.py3-none-any.whl (11 kB)
Collecting tzdata>=2022.1
  Downloading tzdata-2023.3-py2.py3-none-any.whl (341 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 341.8/341.8 kB 20.9 MB/s eta 0:00:00
Collecting pytz>=2020.1
  Using cached pytz-2023.3-py2.py3-none-any.whl (502 kB)
Collecting exceptiongroup>=1.0.0rc8
  Using cached exceptiongroup-1.1.1-py3-none-any.whl (14 kB)
Collecting tomli>=1.0.0
  Using cached tomli-2.0.1-py3-none-any.whl (12 kB)
Collecting pluggy<2.0,>=0.12
  Using cached pluggy-1.0.0-py2.py3-none-any.whl (13 kB)
Collecting iniconfig
  Using cached iniconfig-2.0.0-py3-none-any.whl (5.9 kB)
Collecting isodate<0.7.0,>=0.6.0
  Using cached isodate-0.6.1-py2.py3-none-any.whl (41 kB)
Collecting charset-normalizer<4,>=2
  Using cached charset_normalizer-3.1.0-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (199 kB)
Collecting certifi>=2017.4.17
  Downloading certifi-2023.5.7-py3-none-any.whl (156 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 157.0/157.0 kB 19.3 MB/s eta 0:00:00
Collecting idna<4,>=2.5
  Using cached idna-3.4-py3-none-any.whl (61 kB)
Collecting threadpoolctl>=2.0.0
  Using cached threadpoolctl-3.1.0-py3-none-any.whl (14 kB)
Collecting jmespath<2.0.0,>=0.7.1
  Downloading jmespath-1.0.1-py3-none-any.whl (20 kB)
Collecting s3transfer<0.7.0,>=0.6.0
  Downloading s3transfer-0.6.1-py3-none-any.whl (79 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 79.8/79.8 kB 9.5 MB/s eta 0:00:00
Collecting botocore<1.30.0,>=1.29.132
  Downloading botocore-1.29.132-py3-none-any.whl (10.7 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 10.7/10.7 MB 50.1 MB/s eta 0:00:00
Collecting sphinxcontrib-jquery!=3.0.0,>=2.0.0
  Downloading sphinxcontrib_jquery-4.1-py2.py3-none-any.whl (121 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 121.1/121.1 kB 14.9 MB/s eta 0:00:00
Collecting nbformat
  Using cached nbformat-5.8.0-py3-none-any.whl (77 kB)
Collecting nbconvert!=5.4
  Downloading nbconvert-7.4.0-py3-none-any.whl (285 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 285.9/285.9 kB 32.8 MB/s eta 0:00:00
Collecting traitlets>=5
  Using cached traitlets-5.9.0-py3-none-any.whl (117 kB)
Collecting notebook>=5.0
  Using cached notebook-6.5.4-py3-none-any.whl (529 kB)
Collecting tornado>=4.5
  Downloading tornado-6.3.1-cp38-abi3-manylinux_2_5_x86_64.manylinux1_x86_64.manylinux_2_17_x86_64.manylinux2014_x86_64.whl (426 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 426.8/426.8 kB 43.6 MB/s eta 0:00:00
Collecting urllib3>=1.24.0
  Using cached urllib3-1.26.15-py2.py3-none-any.whl (140 kB)
Collecting MarkupSafe>=2.0
  Downloading MarkupSafe-2.1.2-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (25 kB)
Collecting jupyter-core>=4.7
  Using cached jupyter_core-5.3.0-py3-none-any.whl (93 kB)
Collecting jupyterlab-pygments
  Using cached jupyterlab_pygments-0.2.2-py2.py3-none-any.whl (21 kB)
Collecting nbclient>=0.5.0
  Downloading nbclient-0.7.4-py3-none-any.whl (73 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 73.1/73.1 kB 8.7 MB/s eta 0:00:00
Collecting pandocfilters>=1.4.1
  Using cached pandocfilters-1.5.0-py2.py3-none-any.whl (8.7 kB)
Collecting mistune<3,>=2.0.3
  Using cached mistune-2.0.5-py2.py3-none-any.whl (24 kB)
Collecting defusedxml
  Using cached defusedxml-0.7.1-py2.py3-none-any.whl (25 kB)
Collecting tinycss2
  Using cached tinycss2-1.2.1-py3-none-any.whl (21 kB)
Collecting beautifulsoup4
  Using cached beautifulsoup4-4.12.2-py3-none-any.whl (142 kB)
Collecting bleach
  Using cached bleach-6.0.0-py3-none-any.whl (162 kB)
Collecting jsonschema>=2.6
  Using cached jsonschema-4.17.3-py3-none-any.whl (90 kB)
Collecting fastjsonschema
  Using cached fastjsonschema-2.16.3-py3-none-any.whl (23 kB)
Collecting argon2-cffi
  Using cached argon2_cffi-21.3.0-py3-none-any.whl (14 kB)
Collecting prometheus-client
  Using cached prometheus_client-0.16.0-py3-none-any.whl (122 kB)
Collecting terminado>=0.8.3
  Using cached terminado-0.17.1-py3-none-any.whl (17 kB)
Collecting pyzmq>=17
  Using cached pyzmq-25.0.2-cp310-cp310-manylinux_2_28_x86_64.whl (1.1 MB)
Collecting ipython-genutils
  Using cached ipython_genutils-0.2.0-py2.py3-none-any.whl (26 kB)
Collecting Send2Trash>=1.8.0
  Downloading Send2Trash-1.8.2-py3-none-any.whl (18 kB)
Collecting nbclassic>=0.4.7
  Downloading nbclassic-1.0.0-py3-none-any.whl (10.0 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 10.0/10.0 MB 57.4 MB/s eta 0:00:00
Collecting ipykernel
  Downloading ipykernel-6.23.0-py3-none-any.whl (150 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 150.6/150.6 kB 18.3 MB/s eta 0:00:00
Collecting jupyter-client>=5.3.4
  Downloading jupyter_client-8.2.0-py3-none-any.whl (103 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 103.2/103.2 kB 12.6 MB/s eta 0:00:00
Collecting nest-asyncio>=1.5
  Using cached nest_asyncio-1.5.6-py3-none-any.whl (5.2 kB)
Collecting littleutils
  Using cached littleutils-0.2.2-py3-none-any.whl
Collecting mpmath>=0.19
  Downloading mpmath-1.3.0-py3-none-any.whl (536 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 536.2/536.2 kB 46.9 MB/s eta 0:00:00
Collecting attrs>=17.4.0
  Downloading attrs-23.1.0-py3-none-any.whl (61 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 61.2/61.2 kB 7.2 MB/s eta 0:00:00
Collecting pyrsistent!=0.17.0,!=0.17.1,!=0.17.2,>=0.14.0
  Using cached pyrsistent-0.19.3-py3-none-any.whl (57 kB)
Collecting platformdirs>=2.5
  Downloading platformdirs-3.5.0-py3-none-any.whl (15 kB)
Collecting jupyter-server>=1.8
  Using cached jupyter_server-2.5.0-py3-none-any.whl (366 kB)
Collecting notebook-shim>=0.2.3
  Downloading notebook_shim-0.2.3-py3-none-any.whl (13 kB)
Collecting ptyprocess
  Using cached ptyprocess-0.7.0-py2.py3-none-any.whl (13 kB)
Collecting argon2-cffi-bindings
  Using cached argon2_cffi_bindings-21.2.0-cp36-abi3-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (86 kB)
Collecting soupsieve>1.2
  Downloading soupsieve-2.4.1-py3-none-any.whl (36 kB)
Collecting webencodings
  Using cached webencodings-0.5.1-py2.py3-none-any.whl (11 kB)
Collecting ipython>=7.23.1
  Downloading ipython-8.13.2-py3-none-any.whl (797 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 797.7/797.7 kB 44.9 MB/s eta 0:00:00
Collecting matplotlib-inline>=0.1
  Using cached matplotlib_inline-0.1.6-py3-none-any.whl (9.4 kB)
Collecting comm>=0.1.1
  Using cached comm-0.1.3-py3-none-any.whl (6.6 kB)
Collecting debugpy>=1.6.5
  Using cached debugpy-1.6.7-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (3.0 MB)
Collecting backcall
  Using cached backcall-0.2.0-py2.py3-none-any.whl (11 kB)
Collecting pickleshare
  Using cached pickleshare-0.7.5-py2.py3-none-any.whl (6.9 kB)
Collecting stack-data
  Using cached stack_data-0.6.2-py3-none-any.whl (24 kB)
Collecting jedi>=0.16
  Using cached jedi-0.18.2-py2.py3-none-any.whl (1.6 MB)
Collecting prompt-toolkit!=3.0.37,<3.1.0,>=3.0.30
  Using cached prompt_toolkit-3.0.38-py3-none-any.whl (385 kB)
Collecting decorator
  Using cached decorator-5.1.1-py3-none-any.whl (9.1 kB)
Collecting pexpect>4.3
  Using cached pexpect-4.8.0-py2.py3-none-any.whl (59 kB)
Collecting jupyter-server-terminals
  Using cached jupyter_server_terminals-0.4.4-py3-none-any.whl (13 kB)
Collecting anyio>=3.1.0
  Using cached anyio-3.6.2-py3-none-any.whl (80 kB)
Collecting jupyter-events>=0.4.0
  Using cached jupyter_events-0.6.3-py3-none-any.whl (18 kB)
Collecting websocket-client
  Using cached websocket_client-1.5.1-py3-none-any.whl (55 kB)
Collecting cffi>=1.0.1
  Downloading cffi-1.15.1-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (441 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 441.8/441.8 kB 39.2 MB/s eta 0:00:00
Collecting sniffio>=1.1
  Using cached sniffio-1.3.0-py3-none-any.whl (10 kB)
Collecting pycparser
  Downloading pycparser-2.21-py2.py3-none-any.whl (118 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 118.7/118.7 kB 9.6 MB/s eta 0:00:00
Collecting parso<0.9.0,>=0.8.0
  Using cached parso-0.8.3-py2.py3-none-any.whl (100 kB)
Collecting rfc3986-validator>=0.1.1
  Using cached rfc3986_validator-0.1.1-py2.py3-none-any.whl (4.2 kB)
Collecting python-json-logger>=2.0.4
  Using cached python_json_logger-2.0.7-py3-none-any.whl (8.1 kB)
Collecting rfc3339-validator
  Using cached rfc3339_validator-0.1.4-py2.py3-none-any.whl (3.5 kB)
Collecting wcwidth
  Using cached wcwidth-0.2.6-py2.py3-none-any.whl (29 kB)
Collecting pure-eval
  Using cached pure_eval-0.2.2-py3-none-any.whl (11 kB)
Collecting executing>=1.2.0
  Using cached executing-1.2.0-py2.py3-none-any.whl (24 kB)
Collecting asttokens>=2.1.0
  Using cached asttokens-2.2.1-py2.py3-none-any.whl (26 kB)
Collecting isoduration
  Using cached isoduration-20.11.0-py3-none-any.whl (11 kB)
Collecting uri-template
  Using cached uri_template-1.2.0-py3-none-any.whl (10 kB)
Collecting jsonpointer>1.13
  Using cached jsonpointer-2.3-py2.py3-none-any.whl (7.8 kB)
Collecting fqdn
  Using cached fqdn-1.5.1-py3-none-any.whl (9.1 kB)
Collecting webcolors>=1.11
  Using cached webcolors-1.13-py3-none-any.whl (14 kB)
Collecting arrow>=0.15.0
  Using cached arrow-1.2.3-py3-none-any.whl (66 kB)
Building wheels for collected packages: sphinxemoji, lit
  Building wheel for sphinxemoji (setup.py): started
  Building wheel for sphinxemoji (setup.py): finished with status 'done'
  Created wheel for sphinxemoji: filename=sphinxemoji-0.2.0-py3-none-any.whl size=44410 sha256=56bf750ff106555eb3ed88a7ae9a9ddd9f54a9a3c1d92de4f5d9fd2670dee27d
  Stored in directory: /home/tingyuw/.cache/pip/wheels/64/a4/96/10ec59245ae6f2b45d4509261572f3d6a0db4a7c00d7a6d4c6
  Building wheel for lit (setup.py): started
  Building wheel for lit (setup.py): finished with status 'done'
  Created wheel for lit: filename=lit-16.0.3-py3-none-any.whl size=88174 sha256=397cd54dd6e7d8bc282bf5442e1f9a77f1cb0e870944f6b3f050c6443495bac1
  Stored in directory: /home/tingyuw/.cache/pip/wheels/d6/81/1c/a49ba782377339294cc45c9899927b61a92e58d6ad3ac942f7
Successfully built sphinxemoji lit
Installing collected packages: webencodings, wcwidth, snowballstemmer, pytz, pure-eval, ptyprocess, pickleshare, nose, mpmath, mistune, littleutils, lit, ipython-genutils, fastjsonschema, executing, cmake, backcall, websocket-client, webcolors, urllib3, uri-template, tzdata, typing-extensions, traitlets, tqdm, tornado, tomli, tinycss2, threadpoolctl, sympy, sphinxcontrib-serializinghtml, sphinxcontrib-qthelp, sphinxcontrib-jsmath, sphinxcontrib-htmlhelp, sphinxcontrib-devhelp, sphinxcontrib-applehelp, soupsieve, sniffio, six, Send2Trash, rfc3986-validator, regex, pyzmq, pyyaml, python-json-logger, pyrsistent, pyparsing, Pygments, pycparser, psutil, prompt-toolkit, prometheus-client, pluggy, platformdirs, pillow, pexpect, parso, pandocfilters, packaging, numpy, networkx, nest-asyncio, MarkupSafe, kiwisolver, jupyterlab-pygments, jsonpointer, joblib, jmespath, iniconfig, imagesize, idna, fqdn, fonttools, filelock, exceptiongroup, docutils, defusedxml, decorator, debugpy, cython, cycler, click, charset-normalizer, certifi, babel, attrs, alabaster, terminado, scipy, rfc3339-validator, requests, python-dateutil, pytest, pydantic, pyarrow, nltk, matplotlib-inline, jupyter-core, jsonschema, jinja2, jedi, isodate, contourpy, comm, cffi, bleach, beautifulsoup4, asttokens, anyio, stack-data, sphinx, scikit-learn, rdflib, pandas, outdated, nbformat, matplotlib, jupyter-server-terminals, jupyter-client, botocore, arrow, argon2-cffi-bindings, sphinxemoji, sphinxcontrib-jquery, sphinx-gallery, sphinx_copybutton, seaborn, s3transfer, nbclient, isoduration, ipython, argon2-cffi, sphinx_rtd_theme, nbconvert, ipykernel, boto3, nbsphinx, jupyter-events, nbsphinx-link, jupyter-server, notebook-shim, nbclassic, notebook, jupyter_http_over_ws, triton, torch, torchmetrics, ogb
Successfully installed MarkupSafe-2.1.2 Pygments-2.15.1 Send2Trash-1.8.2 alabaster-0.7.13 anyio-3.6.2 argon2-cffi-21.3.0 argon2-cffi-bindings-21.2.0 arrow-1.2.3 asttokens-2.2.1 attrs-23.1.0 babel-2.12.1 backcall-0.2.0 beautifulsoup4-4.12.2 bleach-6.0.0 boto3-1.26.132 botocore-1.29.132 certifi-2023.5.7 cffi-1.15.1 charset-normalizer-3.1.0 click-8.1.3 cmake-3.26.3 comm-0.1.3 contourpy-1.0.7 cycler-0.11.0 cython-0.29.34 debugpy-1.6.7 decorator-5.1.1 defusedxml-0.7.1 docutils-0.17.1 exceptiongroup-1.1.1 executing-1.2.0 fastjsonschema-2.16.3 filelock-3.12.0 fonttools-4.39.4 fqdn-1.5.1 idna-3.4 imagesize-1.4.1 iniconfig-2.0.0 ipykernel-6.23.0 ipython-8.13.2 ipython-genutils-0.2.0 isodate-0.6.1 isoduration-20.11.0 jedi-0.18.2 jinja2-3.1.2 jmespath-1.0.1 joblib-1.2.0 jsonpointer-2.3 jsonschema-4.17.3 jupyter-client-8.2.0 jupyter-core-5.3.0 jupyter-events-0.6.3 jupyter-server-2.5.0 jupyter-server-terminals-0.4.4 jupyter_http_over_ws-0.0.8 jupyterlab-pygments-0.2.2 kiwisolver-1.4.4 lit-16.0.3 littleutils-0.2.2 matplotlib-3.7.1 matplotlib-inline-0.1.6 mistune-2.0.5 mpmath-1.3.0 nbclassic-1.0.0 nbclient-0.7.4 nbconvert-7.4.0 nbformat-5.8.0 nbsphinx-0.9.1 nbsphinx-link-1.3.0 nest-asyncio-1.5.6 networkx-3.1 nltk-3.8.1 nose-1.3.7 notebook-6.5.4 notebook-shim-0.2.3 numpy-1.24.3 ogb-1.3.6 outdated-0.2.2 packaging-23.1 pandas-2.0.1 pandocfilters-1.5.0 parso-0.8.3 pexpect-4.8.0 pickleshare-0.7.5 pillow-9.5.0 platformdirs-3.5.0 pluggy-1.0.0 prometheus-client-0.16.0 prompt-toolkit-3.0.38 psutil-5.9.5 ptyprocess-0.7.0 pure-eval-0.2.2 pyarrow-12.0.0 pycparser-2.21 pydantic-1.10.7 pyparsing-3.0.9 pyrsistent-0.19.3 pytest-7.3.1 python-dateutil-2.8.2 python-json-logger-2.0.7 pytz-2023.3 pyyaml-6.0 pyzmq-25.0.2 rdflib-6.3.2 regex-2023.5.5 requests-2.30.0 rfc3339-validator-0.1.4 rfc3986-validator-0.1.1 s3transfer-0.6.1 scikit-learn-1.2.2 scipy-1.10.1 seaborn-0.12.2 six-1.16.0 sniffio-1.3.0 snowballstemmer-2.2.0 soupsieve-2.4.1 sphinx-4.2.0 sphinx-gallery-0.13.0 sphinx_copybutton-0.5.2 sphinx_rtd_theme-1.2.0 sphinxcontrib-applehelp-1.0.4 sphinxcontrib-devhelp-1.0.2 sphinxcontrib-htmlhelp-2.0.1 sphinxcontrib-jquery-4.1 sphinxcontrib-jsmath-1.0.1 sphinxcontrib-qthelp-1.0.3 sphinxcontrib-serializinghtml-1.1.5 sphinxemoji-0.2.0 stack-data-0.6.2 sympy-1.12 terminado-0.17.1 threadpoolctl-3.1.0 tinycss2-1.2.1 tomli-2.0.1 torch-2.0.0+cu117 torchmetrics-0.11.4 tornado-6.3.1 tqdm-4.65.0 traitlets-5.9.0 triton-2.0.0 typing-extensions-4.5.0 tzdata-2023.3 uri-template-1.2.0 urllib3-1.26.15 wcwidth-0.2.6 webcolors-1.13 webencodings-0.5.1 websocket-client-1.5.1

done
#
# To activate this environment, use
#
#     $ conda activate dgl-dev-gpu
#
# To deactivate an active environment, use
#
#     $ conda deactivate

Retrieving notices: ...working... done
➜  dgl git:(master) ✗ conda activate dgl-dev-gpu
(dgl-dev-gpu) ➜  dgl git:(master) ✗ conda install -c nvidia -c conda-forge -c rapidsai pylibcugraphops=23.04
Collecting package metadata (current_repodata.json): done
Solving environment: failed with initial frozen solve. Retrying with flexible solve.
Solving environment: failed with repodata from current_repodata.json, will retry with next repodata source.
Collecting package metadata (repodata.json): done
Solving environment: done


==> WARNING: A newer version of conda exists. <==
  current version: 22.9.0
  latest version: 23.3.1

Please update conda by running

    $ conda update -n base -c defaults conda



## Package Plan ##

  environment location: /home/tingyuw/miniconda3/envs/dgl-dev-gpu

  added / updated specs:
    - pylibcugraphops=23.04


The following NEW packages will be INSTALLED:

  cuda-profiler-api  nvidia/linux-64::cuda-profiler-api-11.8.86-0 None
  cudatoolkit        nvidia/linux-64::cudatoolkit-11.7.0-hd8887f6_10 None
  fmt                conda-forge/linux-64::fmt-9.1.0-h924138e_0 None
  libcublas          nvidia/linux-64::libcublas-11.11.3.6-0 None
  libcublas-dev      nvidia/linux-64::libcublas-dev-11.11.3.6-0 None
  libcugraphops      nvidia/linux-64::libcugraphops-23.04.00-cuda11_230412_ga76892e3_0 None
  libcurand          nvidia/linux-64::libcurand-10.3.0.86-0 None
  libcurand-dev      nvidia/linux-64::libcurand-dev-10.3.0.86-0 None
  libcusolver        nvidia/linux-64::libcusolver-11.4.1.48-0 None
  libcusolver-dev    nvidia/linux-64::libcusolver-dev-11.4.1.48-0 None
  libcusparse        nvidia/linux-64::libcusparse-11.7.5.86-0 None
  libcusparse-dev    nvidia/linux-64::libcusparse-dev-11.7.5.86-0 None
  libraft-headers    rapidsai/linux-64::libraft-headers-23.04.01-cuda11_230421_gdc800d6f_0 None
  librmm             rapidsai/linux-64::librmm-23.04.01-cuda11_230421_geab50f46_0 None
  llvm-openmp        conda-forge/linux-64::llvm-openmp-12.0.1-h4bd325d_1 None
  pylibcugraphops    nvidia/linux-64::pylibcugraphops-23.04.00-cuda11_py310_230412_ga76892e3_0 None
  python_abi         conda-forge/linux-64::python_abi-3.10-2_cp310 None
  spdlog             conda-forge/linux-64::spdlog-1.11.0-h9b3ece8_1 None

The following packages will be REMOVED:

  libgomp-11.2.0-h1234567_1

The following packages will be UPDATED:

  ca-certificates    pkgs/main::ca-certificates-2023.01.10~ --> conda-forge::ca-certificates-2023.5.7-hbcca054_0 None
  libgcc-ng          pkgs/main::libgcc-ng-11.2.0-h1234567_1 --> conda-forge::libgcc-ng-12.2.0-h65d4601_19 None
  libstdcxx-ng       pkgs/main::libstdcxx-ng-11.2.0-h12345~ --> conda-forge::libstdcxx-ng-12.2.0-h46fd767_19 None

The following packages will be SUPERSEDED by a higher-priority channel:

  _libgcc_mutex           pkgs/main::_libgcc_mutex-0.1-main --> conda-forge::_libgcc_mutex-0.1-conda_forge None
  _openmp_mutex          pkgs/main::_openmp_mutex-5.1-1_gnu --> conda-forge::_openmp_mutex-4.5-2_kmp_llvm None
  openssl              pkgs/main::openssl-1.1.1t-h7f8727e_0 --> conda-forge::openssl-1.1.1t-h0b41bf4_0 None


Proceed ([y]/n)? y

Preparing transaction: done
Verifying transaction: done
Executing transaction: | By downloading and using the CUDA Toolkit conda packages, you accept the terms and conditions of the CUDA End User License Agreement (EULA): https://docs.nvidia.com/cuda/eula/index.html

done
Retrieving notices: ...working... done


(dgl-dev-gpu) ➜  python git:(master) ✗ python -c "import pylibcugraphops;print(pylibcugraphops.__version__)" 
23.04.00


```

**Comment by Rhett-Ying** [comment]
@tingyu66 do I need to build DGL from latest master branch source code?

**Comment by tingyu66** [comment]
> @tingyu66 do I need to build DGL from latest master branch source code?

Yes, as long as your version includes the changes in #5540.

**Comment by Rhett-Ying** [comment]
As #5540 is not included in `1.1.0`.  we need to try with latest nightly built DGL. Unfortunately, current DGL nightly build is down, so @paoxiaode you need to build DGL from source(latest master branch), refer to [here](https://docs.dgl.ai/install/index.html#linux). Once latest DGL nightly build is ready, I will get back here.

**Comment by jermainewang** [comment]
Nightly build should be back online. @paoxiaode please try it. I will close this for now, but feel free to reopen if the issue still exists.
